src/Pos/TxHistory.hs

```diff
@@ -0,0 +1,241 @@
+{-# LANGUAGE CPP                  #-}
+{-# LANGUAGE InstanceSigs         #-}
+{-# LANGUAGE RankNTypes           #-}
+{-# LANGUAGE ScopedTypeVariables  #-}
+{-# LANGUAGE TemplateHaskell      #-}
```

Comment:
I didn't notice `TemplateHaskell` usages. Maybe it's redundant extension here or maybe I didn't pay enough attention. As well as `InstanceSigs`